core/src/main/java/com/linecorp/armeria/common/stream/DefaultHttpDeframer.java

```diff
@@ -16,75 +16,83 @@
 
 package com.linecorp.armeria.common.stream;
 
+import static com.linecorp.armeria.common.stream.SubscriptionOption.NOTIFY_CANCELLATION;
+import static com.linecorp.armeria.common.stream.SubscriptionOption.WITH_POOLED_OBJECTS;
 import static java.util.Objects.requireNonNull;
 
-import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
-import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
 import java.util.function.Function;
 
 import javax.annotation.Nullable;
 
+import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 
 import com.linecorp.armeria.common.HttpData;
 import com.linecorp.armeria.common.HttpHeaders;
 import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.RequestHeaders;
 import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.annotation.UnstableApi;
 import com.linecorp.armeria.common.util.Exceptions;
 
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.util.concurrent.EventExecutor;
 
 /**
- * The default {@link HttpDeframer} implementation.
+ * The default HTTP deframer implementation.
  */
-final class DefaultHttpDeframer<T>
-        extends DefaultStreamMessage<T>
-        implements HttpDeframer<T>, HttpDeframerOutput<T> {
+@UnstableApi
+public final class DefaultHttpDeframer<T> extends DefaultStreamMessage<T> implements HttpDeframerOutput<T> {
```

Comment:
Shouldn't we hide this from the public API?

Reply:
`DefaultHttpDeframer` uses/override package-private methods such as `demand()` and `cancel()`.
I had the same problem when working on multipart. I added `ListenableStreamMessage` that add hooks by wrapping `Subscriber` and `Subsription`.
https://github.com/line/armeria/pull/2894/files#diff-3dcae84e4c76922eb2d75e9d9bdd86904d33635a2c5a43226c7b6daf8c0a69b3

Reply:
`DefaultHttpDeframer` is moved to `internal` package by adding `demand()` method to `StreamMessage`.